### PR TITLE
feat: Move transaction webhooks to Redis

### PR DESCRIPTION
**Pull request by arcoraven**
## PR-Codex overview
This PR introduces changes related to handling webhooks and transactions. It refactors webhook handling, updates transaction schemas, and improves webhook processing efficiency.

### Detailed summary
- Refactored webhook handling logic
- Updated transaction schemas
- Improved webhook processing efficiency
- Renamed functions for clarity

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/worker/tasks/sendWebhookWorker.ts [hunk 1]
```diff
@@ -11,36 +16,53 @@ import {
...
+      break;
+    }
+
+    default:
+      console.log(`Unknown webhook type: ${(data as any).type}`);
+      return;
   }
 
-  if (resp && !resp.ok) {
+  if (!resp.ok) {
```
Comment (edwardysun) [inline src/worker/tasks/sendWebhookWorker.ts:65]: resp has type `WebhookResponse | undefined`. would this change cause unexpected behavior if resp is undefined?

**Review by edwardysun — APPROVED** [review]

File: src/db/transactions/getTxByIds.ts [hunk 2]
```diff
@@ -1,30 +1,12 @@
...
-  pgtx?: PrismaTransaction;
-}
 
-export const getTxByIds = async ({
-  queueIds,
-}: GetTxByIdsParams): Promise<
-  Static<typeof transactionResponseSchema>[] | null
-> => {
-  const tx = await prisma.transactions.findMany({
+export const getTransactionsByQueueIds = async (
```
Comment (arcoraven) [inline src/db/transactions/getTxByIds.ts:4]: Simplified this to just return DB results.

File: src/db/webhooks/getAllWebhooks.ts [hunk 3]
```diff
@@ -3,6 +3,9 @@ import { prisma } from "../client";
 
 export const getAllWebhooks = async (): Promise<Webhooks[]> => {
   return await prisma.webhooks.findMany({
+    where: {
```
Comment (arcoraven) [inline src/db/webhooks/getAllWebhooks.ts:6]: We never care about deleted webhooks so omit it in the query to simplify business logic.

File: src/server/schemas/transaction/index.ts [hunk 4]
```diff
@@ -198,3 +200,23 @@ export enum TransactionStatus {
   // Tx was cancelled and will not be re-attempted.
   Cancelled = "cancelled",
 }
+
+export const toTransactionSchema = (
```
Comment (arcoraven) [inline src/server/schemas/transaction/index.ts:204]: Maps the internal DB "transaction" to the external "transaction" schema.

File: src/worker/queues/sendWebhookQueue.ts [hunk 5]
```diff
@@ -88,3 +115,36 @@ const getContractSubscriptionWebhookIdempotencyKey = (args: {
...
+
+const enqueueTransactionWebhook = async (
+  data: EnqueueTransactionWebhookData,
+) => {
+  if (!_queue) return;
+
+  const webhooks = [
+    ...(await getWebhooksByEventType(WebhooksEventTypes.ALL_TX)),
+    ...(await getWebhooksByEventType(data.type)),
+  ];
```
Comment (arcoraven) [inline src/worker/queues/sendWebhookQueue.ts:131]: Get all webhooks that match the exact type or `ALL_TX` which cares about all event types.

File: src/worker/tasks/sendWebhookWorker.ts [hunk 6]
```diff
@@ -52,7 +70,7 @@ const handler: Processor<any, void, string> = async (job: Job<string>) => {
 let _worker: Worker | null = null;
 if (redis) {
   _worker = new Worker(SEND_WEBHOOK_QUEUE_NAME, handler, {
-    concurrency: 1,
+    concurrency: 10,
```
Comment (arcoraven) [inline src/worker/tasks/sendWebhookWorker.ts:73]: Increase concurrency now that there's more webhooks to send. Webhook calls are independent.

**Pull request merged**